Functions returning UArray* must not return the Io nil object (GCC 14 error)

--- a/libs/iovm/source/IoFile.c	2026-09-25 23:33:21.511231020 -0700
+++ b/libs/iovm/source/IoFile.c	2026-09-25 23:33:21.521154622 -0700
@@ -812,7 +812,10 @@
     size_t length = IoMessage_locals_sizetArgAt_(m, locals, 0);
     UArray *ba = UArray_new();
     IoFile_assertOpen(self, locals, m);
-    if (IOSTATE->errorRaised) return IONIL(self);
+    if (IOSTATE->errorRaised) {
+        UArray_free(ba);
+        return NULL;
+    }
 
     UArray_readNumberOfItems_fromCStream_(ba, length, DATA(self)->stream);
 
--- a/libs/iovm/source/IoObject.c	2026-09-25 23:33:52.596740333 -0700
+++ b/libs/iovm/source/IoObject.c	2026-09-25 23:33:52.606119197 -0700
@@ -1913,14 +1913,20 @@
 UArray *IoObject_rawGetUArraySlot(IoObject *self, IoObject *locals,
                                   IoMessage *m, IoSymbol *slotName) {
     IoSeq *seq = IoObject_getSlot_(self, slotName);
-    IOASSERT(ISSEQ(seq), CSTRING(slotName));
+    if (!ISSEQ(seq)) {
+        IoState_error_(IOSTATE, m, "Io Assertion '%s'", CSTRING(slotName));
+        return NULL;
+    }
     return IoSeq_rawUArray(seq);
 }
 
 UArray *IoObject_rawGetMutableUArraySlot(IoObject *self, IoObject *locals,
                                          IoMessage *m, IoSymbol *slotName) {
     IoSeq *seq = IoObject_getSlot_(self, slotName);
-    IOASSERT(ISSEQ(seq), CSTRING(slotName));
+    if (!ISSEQ(seq)) {
+        IoState_error_(IOSTATE, m, "Io Assertion '%s'", CSTRING(slotName));
+        return NULL;
+    }
     return IoSeq_rawUArray(seq);
 }
 
